util/parser.go

```diff
@@ -73,6 +76,23 @@ func FormatTimeToString(inputTime time.Time, format string) string {
 	return inputTime.Format(format)
 }
 
-func DateTimeToRecordFormat(inputTime time.Time) *string {
-	return StringValueOrNil(FormatTimeToString(inputTime, RecordDateTimeFormat))
+func DateTimeToRFC3339(inputTime interface{}) string {
+	var timeField time.Time
+	switch v := inputTime.(type) {
+
+	case *time.Time:
+		if reflect.ValueOf(inputTime).IsNil() {
```

Comment:
Luckily you don't have to use reflection here, you can just have a `case nil:` since an interface{} can be the nil type.